tests/countries/test_nigeria.py

```diff
@@ -19,7 +19,10 @@
 class TestNigeria(CommonCountryTests, TestCase):
     @classmethod
     def setUpClass(cls):
-        super().setUpClass(Nigeria, years=range(1979, 2050))
+        years = range(1979, 2050)
+        super().setUpClass(Nigeria, years=years, years_non_observed=years)
+        cls.full_range = years
```

Comment:
👍

Reply:
I think, we need to pass years range to `setUpClass`, and let it create `cls.full_range`, `cls.holidays`, `cls.holidays_non_observed`...

Reply:
`cls.no_estimated_holidays` - if `test_class` inherited from `IslamicHolidays` (or even `EasternCalendarHolidays`),
`cls.subdiv_holidays` - if `test_class` have subdivisions...

Reply:
To make this easier for review - I'm going to do one for `years_no_estimated` first (with the migration for other entities in another PR after this got merged in), then a separate one for `years_subdiv`